tests/integration/crunchers/test_cruncher_equal_quantile_walk.py

```diff
@@ -182,7 +182,71 @@ def test_numerical_relationship(self, test_db):
 
         expected = np.quantile(ys, quant_of_y)
 
-        assert all(crunched["value"].values == expected)
+        assert np.allclose(crunched["value"].values, expected)
+
+    @pytest.mark.parametrize("smoothing", [True, False])
+    def test_weighting_results(self, test_db, smoothing):
+        # Ensure that duplicating data and giving it a weight of 1/2 makes little
+        # difference to the results
+
+        # First check that giving a weight of 1 does nothing.
+        large_db = IamDataFrame(self.large_db.copy())
+        large_db = self._adjust_time_style_to_match(large_db, test_db)
+        to_infill = test_db.filter(
+            **{test_db.time_col: large_db[large_db.time_col][0]}
+        ).data
+        to_infill["value"].iloc[-1] = 3.7
+        to_infill = IamDataFrame(to_infill)
+        lead = ["Emissions|CO2"]
+        follow = "Emissions|CH4"
+        tcruncher = self.tclass(large_db)
+        normal_results = tcruncher.derive_relationship(
+            variable_follower=follow, variable_leaders=lead, smoothing=smoothing
+        )(to_infill)
+        weight_1 = {(_ma, _sb): 1, (_ma, _sa): 1}
+        norm_weight_results = tcruncher.derive_relationship(
+            variable_follower=follow,
+            variable_leaders=lead,
+            weighting=weight_1,
+            smoothing=smoothing,
+        )(to_infill)
+        assert normal_results.equals(norm_weight_results)
+
+        # Compare these with a run where we have a duplicated datapoint.
+        new_scen = "new_scen"
+        duplicate = large_db.filter(model=_mb, scenario=_sd)
+        duplicate.rename({"scenario": {_sd: new_scen}}, inplace=True)
+        larger_db = large_db.append(duplicate)
+        tcruncher_dup = self.tclass(larger_db)
+        weights = {(_mb, _sd): 0.5, (_mb, new_scen): 0.5}
+        weighted_results = tcruncher_dup.derive_relationship(
+            variable_follower=follow,
+            variable_leaders=lead,
+            weighting=weights,
+            smoothing=smoothing,
+        )(to_infill)
+        if smoothing:
+            np.allclose(
+                normal_results.data.value, weighted_results.data.value, rtol=0.002
+            )
+        else:
+            # The first points are precisely equal but the final point is interpolated
+            # slightly differently
+            assert normal_results.data.iloc[:3].equals(weighted_results.data.iloc[:3])
```

Comment:
Why is `:3` hard-coded?

Reply:
As opposed to ":-1"? We're testing the first three components of a four-component item.

Reply:
Yep, given the comment, `:-1` would be clearer.

Do I understand correctly that the final point changes because of the duplications and combinations used in the test but these don't affect the other points?

Reply:
Yes, without the smoothing the way the interpolation works slightly throws off the arithmetic you might expect here (duplicate at half weight means no change) so it's not quite equal.